### ISSUE TITLE
Server side filter and limit fields does not work together.

### ISSUE THREAD

**Issue by DavidSichau** [aslagle/reactive-table#285]
I have problems to understand how the server side filter and limitation of fields work together.

I have the following publication:
```
ReactiveTable.publish("user-managment",
    function () {
        if(this.userId && Roles.isAdminMethod(this.userId)) {
            return Meteor.users;
        } else {
            return [];
        }
    }
);
```
This works fine however it publishes all fields.
I then tried to limit the fields:

```
ReactiveTable.publish("user-managment",
    function () {
        if(this.userId && Roles.isAdminMethod(this.userId)) {
            return Meteor.users;
        } else {
            return [];
        }
    }, {}, {fields: {emails: 1, profile: 1, roles: 1, status: 1, createdAt: 1}}
);
```

This also works, however I now cannot filter anymore.

**Comment by aslagle** [comment]
If you're limiting the fields, you'll only be able to filter on the fields you publish, for security reasons.

Otherwise, it should work - which fields are you trying to filter on?

**Comment by DavidSichau** [comment]
I only filter on published fields, here is the setting file I use:

```
  settings: function () {
        return {
            collection: "user-managment",
            rowsPerPage: 10,
            useFontAwesome:true,
            showFilter: true,
            fields: [
                {'key': 'profile.firstName', label: function() {return TAPi18n.__("_user_firstName")}},
                {'key': 'profile.lastName', label: function() {return TAPi18n.__("_user_lastName")}},
                {'key': 'emails.0.address', label: function() {return TAPi18n.__("_user_email")}},
                {'key': 'createdAt',
                    sortByValue: true,
                    sortDirection: 'ascending',
                    label: function() {return TAPi18n.__("_user_created");},
                    fn: function(value) {
                        return moment(value).format('lll');
                    }
                },
                {'key': 'status.lastLogin.date',
                    sortByValue: true,
                    sortDirection: 'ascending',
                    label: function() {return TAPi18n.__("_user_lastLogin");},
                    fn: function(value) {
                        return moment(value).format('lll');
                    }
                },
                {'key': 'roles.accessRight.teacher', label: function() {return TAPi18n.__("_user_role_teacher")},
                    sortable: false,
                    fn: function (value, ob) {
                        return checkOrX({
                            check: Roles.userIsInRole(ob, Roles.teacher, Roles.accessRight)
                        });
                    }
                },
                {'key': 'roles.accessRight.editor', label: function() {return TAPi18n.__("_user_role_editor")},
                    sortable: false,
                    fn: function (value, ob) {
                        return checkOrX(Roles.userIsInRole(ob, Roles.editor, Roles.accessRight));
                    }
                },
                {'key': 'roles.accessRight.admin', label: function() {return TAPi18n.__("_user_role_admin")},
                    sortable: false,
                    fn: function (value, ob) {
                        return checkOrX(Roles.userIsInRole(ob, Roles.admin, Roles.accessRight));
                    }
                },
                {'key': 'profile.organisation', label: function() {return TAPi18n.__("_user_organisationStep")},
                    tmpl: Template.userOrganisation
                },
                {'key': 'mangament', sortable: false, label: function() {return ""},
                    tmpl: Template.userEdit
                }
                ]
        };
    }
```

**Comment by aslagle** [comment]
Ah, I think it's because of the nested keys, sorry about that. When I check them for the filtering I'm only checking if the key is an exact match to one that's published, not a child. For now, it might work if you write out the full nested keys to limit to, like {fields: {'emails.0.address': 1, 'profile.firstName': 1, 'profiles.lastName': 1, ...}}

**Comment by DavidSichau** [comment]
@aslagle Thanks that worked. Thanks you also for the great package.

**Comment by DavidSichau** [comment]
One Problem ist with arrays. With objects it works fine however: `{fields: {'emails.0.address':1}}` does not publish the array object.

**Comment by aslagle** [comment]
I haven't tried this, but as a workaround you might be able to put both `emails: 1` and `'emails.0.address': 1` in the fields.

**Comment by DavidSichau** [comment]
This does not work. It throes the following error:

`MinimongoError: both emails.0.address and emails found in fields option, using both of them may trigger unexpected behavior. Did you mean to use only one of them?`

**Comment by aslagle** [comment]
Okay, I think I have a fix for this in the package. I won't have time to test it thoroughly for the next couple weeks, or address any new bugs it causes, so I published it as a beta version (0.8.13-beta1). You'll have to set the version manually and update when I release 0.8.13.

**Comment by DavidSichau** [comment]
The beta1 works perfect. Thanks for the fast response.
